enclave-runtime/src/test/self_sign_cert_tests.rs

```diff
@@ -0,0 +1,35 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+	Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::initialization::enclave_self_signed_certificate;
+use itp_sgx_crypto::Ed25519Seal;
+use itp_sgx_io::StaticSealedIO;
+use sp_core::Pair;
+use std::fs;
+
+const SIGNER: &[u8] = include_bytes!("fixtures/ed25519_key_sealed.bin");
+
+pub fn test_verify_signature_enclave_self_signed_certificate() {
+	fs::write("../bin/ed25519_key_sealed.bin", SIGNER).unwrap();
+	let signing = Ed25519Seal::unseal_from_static_file().unwrap();
```

Comment:
In tests, try to not use file operations. If you need an Ed25519 signing key, you can generate it without a file (there are many examples in tests). With the moving of the logic to the websocket crates, the tests can probably be done with `cargo test`.